### PR TITLE
Alloc zdb_cd_t to fix stack issue

### PR DESCRIPTION
**Pull request by andrewc12**
### Motivation and Context

Pushing a change that is used in OpenZFS on windows upstream.
Fixes a problem since windows apparently has a smaller stack.

### Description

Alloc zdb_cd_t since it is too large for the stack on windows.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: cmd/zdb/zdb.c [hunk 1]
```diff
@@ -6751,9 +6753,12 @@ dump_block_stats(spa_t *spa)
 	if (leaks)
 		return (2);
```
Comment (behlendorf) [inline cmd/zdb/zdb.c:6754]: Looks like your missing a `umem_free()` here.

**Review by behlendorf — COMMENTED** [review]
Dynamically allocating this is fine with me, but I'm a little confused why it's needed since this is all userland code.  Out of curiosity what is the stack size limit on windows?

File: cmd/zdb/zdb.c (hunk 1 above)
Comment (andrewc12) [reply cmd/zdb/zdb.c:6756]: Thank you for picking that up, I also fixed https://github.com/openzfs/zfs/pull/13807/commits/f9687106ff291b2cef65d6a3da97bd09a775b739#diff-0f5313b317e4f72623f33a7c61d0e72115913737e19d910b093dbc7359cc9681R6547

**Comment by andrewc12** [comment]
> Dynamically allocating this is fine with me, but I'm a little confused why it's needed since this is all userland code.

Honestly atm I don't know what exact problem this code fixes.
I chose it because it was one of the first things that conflicts when I rebase the windows branch onto master and it seemed like a nice small first project that would eventually have to get upstreamed.

I've tried reverting this patch and causing a crash, but nothing so far.
I'll do some digging and get back to you.

> Out of curiosity what is the stack size limit on windows?

Without changing it the default that the linker sets for the stack reserve is 1MB
https://docs.microsoft.com/en-us/cpp/build/reference/stack-stack-allocations?redirectedfrom=MSDN&view=msvc-170

**Comment by andrewc12** [comment]
Correction: if I actually use the correct binary with the patch removed, zdb just dies.

**Comment by lundman** [comment]
Yeah it wont even start the .exe with the "too large" variable. 

Also, let me introduce Andrewc12 here, he's been helping me with the Windows port, and doing some of the PR work for me :). 

Do I still review them even if they are "mine" to start with?

**Review by lundman — APPROVED** [review]
Looks fine to me, that lundman guy does some nice work!

**Review by behlendorf — APPROVED** [review]

**Comment by behlendorf** [comment]
@andrewc12 thanks for helping with the windows port!

@lundman I just wanted to make sure you had a chance to look it over, but it sounds like you've done more than that!

**Comment by andrewc12** [comment]
No problem, I've been enjoying working on it.
Thanks for reviewing my first PR.

**Pull request merged**